test/integration/com/unifina/service/PermissionServiceIntegrationSpec.groovy

```diff
@@ -310,4 +342,12 @@ class PermissionServiceIntegrationSpec extends IntegrationSpec {
 		service.canWrite(me, stream)
 		service.canShare(me, stream)
 	}
+
+	void "getPermissionsTo(resource, userish) returns correct UI channel read permissions via associated dashboard"() {
+		service.systemGrantAll(me, dashboard)
+
+		expect:
+		service.getPermissionsTo(uiChannelStream, me).size() == 1
```

Comment:
Minor sanity check improvement: could check that there is a single item AND that single item is a `read` permission